### DISCUSSION TITLE
Liquid simulation?

### DISCUSSION THREAD

**Discussion by ZXMushroom63** [TurboWarp/extensions#555] [Ideas]
While reading the Wikipedia of Box2D, I found that Google had an open source form called "LiquidFun" which adds liquid simulation to Box2D.

I think simulating liquids in Scratch is an interesting idea, and I was wondering if it would be possible to add liquid simulation to the already existing box2d physics extension?

LiquidFun homepage:
https://google.github.io/liquidfun/

Inside LiquidFun:
https://docs.google.com/presentation/d/1fEAb4-lSyqxlVGNPog3G1LZ7UgtvxfRAwR0dwd19G4g

**Comment by BlueDome77** [comment]
It might be too laggy for Turbowarp

**Reply by Flappy25** [reply]
If it's just a simple low res model it may be possible.

**Reply by ZXMushroom63** [reply]
I found a few online demos using PixiJS and LiquidFun, which ran at full 30fps on my cheap Chromebook. I am imagining this is because LiquidFun is transpiled from C++ to JavaScript through Web Assembly, so it runs at somewhat near-native speeds.

The main problem here (I think) is loading the library and getting it to work alongside the turbowarp renderer and box2d physics.

I'll check if the Box2D physics extension still works when replacing the engine with LiquidFun, since LiquidFun is a fork of Box2D.

**Reply by ZXMushroom63** [reply]
Update:
It appears that the latest liquidfun version at this point in time is v1.1.0, which is based on Box2D revision 280 (v2.3.0).
I however, cannot figure out what version of Box2D the extension is using!
All I know is that the licence states 2006-2007.
